contracts/staking/validatorShare/ValidatorShare.sol

```diff
@@ -1,24 +1,59 @@
 pragma solidity ^0.5.2;
-import {ERC20} from "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
 
-import {Registry} from "../../common/Registry.sol";
-import {ValidatorShareStorage} from "./ValidatorShareStorage.sol";
+import {ERC20NonTransferable} from "../../common/tokens/ERC20NonTransferable.sol";
+import {ERC20} from "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
 import {StakingInfo} from "./../StakingInfo.sol";
-import {Lockable} from "../../common/mixin/Lockable.sol";
+import {OwnableLockable} from "../../common/mixin/OwnableLockable.sol";
+import {IStakeManager} from "../stakeManager/IStakeManager.sol";
+import {IValidatorShare} from "./IValidatorShare.sol";
+import {Initializable} from "../../common/mixin/Initializable.sol";
+
+contract ValidatorShare is IValidatorShare, ERC20NonTransferable, OwnableLockable, Initializable {
+    struct Delegator {
+        uint256 shares;
+        uint256 withdrawEpoch;
+    }
+
+    uint256 constant EXCHANGE_RATE_PRECISION = 100;
+    uint256 constant MAX_COMMISION_RATE = 100;
+    uint256 constant REWARD_PRECISION = 10**25;
 
+    StakingInfo public stakingLogger;
+    IStakeManager public stakeManager;
+    uint256 public validatorId;
+    uint256 public validatorRewards;
+    uint256 public commissionRate;
+    //last checkpoint where validator updated commission rate
+    uint256 public lastCommissionUpdate;
+    uint256 public minAmount = 10**18;
+
+    uint256 public totalStake;
+    uint256 public rewardPerShare;
+    uint256 public activeAmount;
+    bool public delegation = true;
+
+    uint256 public withdrawPool;
+    uint256 public withdrawShares;
+
+    mapping(address => uint256) public amountStaked;
+    mapping(address => Delegator) public delegators;
+    mapping(address => uint256) public initalRewardPerShare;
 
-contract ValidatorShare is ValidatorShareStorage {
     modifier onlyValidator() {
-        require(stakeManager.ownerOf(validatorId) == msg.sender);
+        require(stakeManager.ownerOf(validatorId) == msg.sender, "not validator");
         _;
     }
 
-    constructor(
-        address _registry,
-        uint256 _validatorId,
-        address _stakingLogger,
-        address _stakeManager
-    ) public Lockable(_stakeManager) {} // dummy constructor
+    // onlyOwner will prevent this contract from initializing, since it's owner is going to be 0x0 address
+    function initialize(uint256 _validatorId, address _stakingLogger, address _stakeManager) external initializer  {
+        validatorId = _validatorId;
+        stakingLogger = StakingInfo(_stakingLogger);
+        stakeManager = IStakeManager(_stakeManager);
+        _transferOwnership(_stakeManager);
```

Comment:
Since `stakeManager` will be the owner for validator share proxy, only it can update the implementation for validator share, right? 

Shouldn't we allow governance to update proxy?

Reply:
Ownership transfer you highlighted is owner of the ValidatorShare implementation. Implementation lives in the Registry, to update every validator share contract we just need to swap address in the registry itself